src/spotted/data/pending_post.py

```diff
@@ -138,10 +146,23 @@ def get_all(admin_group_id: int, before: datetime | None = None) -> list["Pendin
 
     def save_post(self) -> "PendingPost":
         """Saves the pending_post in the database"""
+        if self.credit_username is None:
+            columns = ("user_id", "u_message_id", "g_message_id", "admin_group_id", "message_date")
+            values = (self.user_id, self.u_message_id, self.g_message_id, self.admin_group_id, self.date)
+        else:
+            columns = ("user_id", "u_message_id", "g_message_id", "admin_group_id", "credit_username", "message_date")
+            values = (
+                self.user_id,
+                self.u_message_id,
+                self.g_message_id,
+                self.admin_group_id,
+                self.credit_username,
+                self.date,
+            )
```

Comment:
This could be simplified with either

```py
        columns = ("user_id", "u_message_id", "g_message_id", "admin_group_id", "message_date")
        values = (self.user_id, self.u_message_id, self.g_message_id, self.admin_group_id, self.date)
        if self.credit_username is not None:
            columns += ("credit_username",)
            values += (self.credit_username,)
```

or, since we know that `self.credit_username` is either the correct string or None,

```py
        columns = ("user_id", "u_message_id", "g_message_id", "admin_group_id", "message_date", "credit_username")
        values = (
            self.user_id,
            self.u_message_id,
            self.g_message_id,
            self.admin_group_id,
            self.date,
            self.credit_username,
        )
```

Reply:
I implemented such if because I do not know how the DBmanager handles the conversion between NULL(DB) <++> None(Python)

By excluding it on the query, I was sure that NoneType was returned instead of "NULL" str.

Reply:
If you want more control, you can use the first option.
[This](https://docs.python.org/3/library/sqlite3.html#sqlite-and-python-types) is the default conversion table of the library.